src/shim/crypto.rs

```diff
@@ -149,6 +154,49 @@ pub fn verify_bls_aggregate(data: &[&[u8]], pub_keys: &[&[u8]], sig: &Signature)
     verify_messages(&bls_sig, data, &pks[..])
 }
 
+/// Returns `String` error if a BLS signature is invalid.
+pub fn verify_bls_sig(signature: &[u8], data: &[u8], addr: &Address) -> Result<(), String> {
```

Comment:
Why have you copied this code?

Reply:
To replace the function from FVM for usage in `src/fil_cns/validation.rs`; I suppose I could have invoked the function without copying the code, but I was being consistent with the `verify_bls_aggregate` function above this.

Reply:
Our copy of `verify_bls_aggregate` is semantically different from `fvm_shared::crypto::signature::verify_bls_aggregate`. If you accidentally used the FVM function, Forest would stop working.

Your copy of `verify_bls_sig` is not different from the FVM version. So let's use the FVM version.